dotcom-rendering/src/web/browser/hydration/doHydration.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import { hydrate, h } from 'preact';
+import { initPerf } from '../initPerf';
+
+/**
+ * This function dynamically imports and then hydrates a specific component in
+ * a specific part of the page
+ *
+ * @param name The name of the component we want to hydrate
+ * @param data The deserialised props we want to use for hydration
+ * @param marker The location on the DOM where the component to hydrate exists
+ */
+export const doHydration = (name: string, data: any, marker: HTMLElement) => {
+	const { start, end } = initPerf(`hydrate-${name}`);
+	start();
+	window
+		.guardianPolyfilledImport(
+			/* webpackInclude: /\.importable\.(tsx|jsx)$/ */
+			`../../components/${name}.importable`,
+		)
+		.then((module) => {
+			hydrate(h(module[name], data), marker);
```

Comment:
This is all a very nice step forwards!

I assume preact will take care of the component lifecycle and any hooks defined server side would run as normal?

Reply:
I need to check this. I am drawn to using preact directly like this because I know that DCR alias react > preact. But it's worth considering what happens in the event that were no longer the case.